native/src/routes/jpal-tracking/components/JpalTracking.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import React, { useEffect, useState } from 'react'
+
+import Caption from '../../../modules/common/components/Caption'
+import type { ThemeType } from 'build-configs/ThemeType'
+import { Switch, Text, TextInput, View } from 'react-native'
+import type { TFunction } from 'react-i18next'
+import type { StateType } from '../../../modules/app/StateType'
+import AppSettings, { defaultSettings } from '../../../modules/settings/AppSettings'
+import type { NavigationPropType, RoutePropType } from '../../../modules/app/constants/NavigationTypes'
+import type { JpalTrackingRouteType } from 'api-client'
+import styled from 'styled-components/native'
+import LayoutContainer from '../../../modules/layout/containers/LayoutContainer'
+
+const ThemedText = styled.Text`
+  display: flex;
+  text-align: left;
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.decorativeFontRegular};
+  padding: 10px 0;
+`
+
+const DescriptionContainer = styled.View`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  padding: 15px 0 5px;
+`
+
+const Input = styled(TextInput)`
+  padding: 15px;
+  border-width: 1px;
+  border-color: ${props => props.theme.colors.themeColor};
+  text-align-vertical: top;
+  height: 50px;
+`
+
+export type PropsType = {|
+  theme: ThemeType,
+  t: TFunction,
+  route: RoutePropType<JpalTrackingRouteType>,
+  navigation: NavigationPropType<JpalTrackingRouteType>
+|}
+
+const JpalTracking = (props: PropsType, state: StateType) => {
+  const [appSettings] = useState(new AppSettings())
```

Comment:
This seems weird, I think keeping the settings in the state should be enough. Although I think you should type `useState<type>`.

Reply:
if just putting it like:

const appSettings = new AppSettings() it get:

ESLint: The 'appSettings' object construction makes the dependencies of useEffect Hook (at line 73) change on every render. To fix this, wrap the initialization of 'appSettings' in its own useMemo() Hook.(react-hooks/exhaustive-deps)

i now changed it to useMemo()

Reply:
You could also just move it out of the component, i.e.:
```
const appSettings = new AppSettings()

const JpalTracking = (props: PropsType) => {
...
```
Then you also don't have to put it into the dependency of the hooks.